msteams-platform/concepts/messaging-extensions/create-extensions.md

```diff
@@ -339,22 +352,130 @@ This used to insert a card into the compose box as a result of a the command. It
+          "content": << Card Payload >>
+        }
+      ]
+    }
+  }
+}
+```
+
+Your message extension will now need to respond to two new types of interactions, `value.botMessagePreviewAction = "send"` and `value.botMessagePreviewAction = "edit"`. Below is an example of the `value` object you will need to process:
+
+```json
+{
+  "name": "composeExtension/submitAction",
+  "type": "invoke",
+  ...
+  "value": {
+    "botMessagePreviewAction": "send",
+    "botActivityPreview": [
+      {
+        "type": "message/card",
+        "attachments": [
+          {
+            "content":
               {
-                "url": "https://placekitten.com/200/200"
+                "type": "AdaptiveCard",
+                "body": [{<<card payload>>}]
               }
-            ]
           }
-        }
+        ],
+        "context": { "theme": "default" }
       }
     ],
-    "type": "result",
-    "attachmentLayout": "list"
+    "conversation": { "id": "19:c366b75791784100b6e8b515fd55b063@thread.skype" },
+    "name": "composeExtension/submitAction",
+    "imdisplayname": "Pranav Smith"
   }
 }
 ```
+
+When responding to the `edit` request you should respond with an adaptive card similar to the original `fetch\task` response, but with the values populated with the information the user has already submitted. When responding to the `submit` request you should send a message to the channel containing the finalized adaptive card. The example below shows how to do this using the [Node.js Teams Bot Builder SDK](https://www.npmjs.com/package/botbuilder-teams).
+
+```typescript
+teamChatConnector.onComposeExtensionSubmitAction((
+    event: builder.IEvent,
+    request: teamBuilder.IComposeExtensionActionCommandRequest,
+    callback: (err: Error, result: any, statusCode: number) => void) => {
+        let invokeValue = (<any> event).value;
+
+        if (invokeValue.botMessagePreviewAction ) {
+            let attachment = invokeValue.botActivityPreview[0].attachments[0];
+
+            if (invokeValue.botMessagePreviewAction === 'send') {
+                let response = { composeExtension: { attachmentLayout: "list", type: "result", attachments: [] } };
+                response.composeExtension.attachments.push(attachment);
+                delete this.recentUserInput;
+                let msg = new builder.Message()
+                    .address(event.address)
+                    .addAttachment(attachment);
+                teamChatConnector.send([msg.toMessage()],
+                    (error) => {
+                        if(error){
+                            //TODO: Handle error and callback
+                        }
+                        else {
+                            callback(null, null, 200);
+                        }
+                    }
+                );
+            }
+
+            else if (invokeValue.botMessagePreviewAction === 'edit') {
+                // Create the card and populate with recentUserInput information
```

Comment:
@robin-liao @aosolis @hamenon-ms I've updated based on the feedback above. Please let me know if the edits I made reflect your desired changes.

Outstanding questions:
- Is the type="message/card" correct for the object in the botActivityPreview JSON?
- What is the suggested error handling for the lambda passed into the teamChatConnector.send function? (I'm not sure if my example code for this is right at all. I don't want to just "..." it as in Robin's example because the callback function is part of it